ods_ci/tests/Tests/400__ods_dashboard/420__model_serving/421__model_serving_gpu.robot

```diff
@@ -0,0 +1,149 @@
+
+
+*** Test Cases ***
+Verify GPU Model Deployment Via UI
+    [Documentation]    Test the deployment of an openvino_ir model on a model server with GPUs attached
+    [Tags]    Sanity    Tier1    Resources-GPU
+    ...    ODS-2214
+    Open Model Serving Home Page
+    Try Opening Create Server
+    Wait for RHODS Dashboard to Load    wait_for_cards=${FALSE}    expected_page=Data science projects
+    Create Data Science Project    title=${PRJ_TITLE}    description=${PRJ_DESCRIPTION}
+    Create S3 Data Connection    project_title=${PRJ_TITLE}    dc_name=model-serving-connection
+    ...            aws_access_key=${S3.AWS_ACCESS_KEY_ID}    aws_secret_access=${S3.AWS_SECRET_ACCESS_KEY}
+    ...            aws_bucket_name=ods-ci-s3
+    # For 1.27 expects a custom runtime called New OVMS Server test to be present with the correct GPU flag
+    Create Model Server    token=${FALSE}    server_name=${RUNTIME_NAME}    no_gpus=1    runtime=New OVMS Server test
+    Verify Displayed GPU Count    server_name=${RUNTIME_NAME}    no_gpus=1
+    Open Model Serving Home Page
+    Serve Model    project_name=${PRJ_TITLE}    model_name=${MODEL_NAME}    framework=openvino_ir    existing_data_connection=${TRUE}  # robocop:disable
+    ...    data_connection_name=model-serving-connection    model_path=vehicle-detection    model_server=${RUNTIME_NAME}
+    ${runtime_pod_name} =    Replace String Using Regexp    string=${RUNTIME_NAME}    pattern=\\s    replace_with=-
+    ${runtime_pod_name} =    Convert To Lower Case    ${runtime_pod_name}
+    Run Keyword And Continue On Failure  Wait Until Keyword Succeeds
+    ...  5 min  10 sec  Verify Openvino Deployment    runtime_name=${runtime_pod_name}
+    Run Keyword And Continue On Failure  Wait Until Keyword Succeeds  5 min  10 sec  Verify Serving Service
+    ${requests} =    Get Container Requests    namespace=${PRJ_TITLE}    
+    ...    label=name=modelmesh-serving-${runtime_pod_name}    container_name=ovms
+    Should Contain    ${requests}    "nvidia.com/gpu": "1"
+    ${node} =    Get Node Pod Is Running On    namespace=${PRJ_TITLE}    label=name=modelmesh-serving-${runtime_pod_name}
+    ${type} =    Get Instance Type Of Node    ${node}
+    Should Be Equal As Strings    ${type}    "g4dn.xlarge"
+    Verify Model Status    ${MODEL_NAME}    success
+    Set Suite Variable    ${MODEL_CREATED}    True
+
+Test Inference Load On GPU
+    [Documentation]    Test the inference load on the GPU after sending random requests to the endpoint
+    [Tags]    Sanity    Tier1    Resources-GPU
+    ...    ODS-2213
+    ${url}=    Get Model Route via UI    ${MODEL_NAME}
+    Send Random Inference Request     endpoint=${url}    no_requests=100
+    # Verify metric DCGM_FI_PROF_GR_ENGINE_ACTIVE goes over 0
+    ${prometheus_route} =    Get OpenShift Prometheus Route
+    ${sa_token} =    Get OpenShift Prometheus Service Account Token
+    ${expression} =    Set Variable    DCGM_FI_PROF_GR_ENGINE_ACTIVE  #&step=1
```

Comment:
is `#&step=1` needed?

Reply:
I'm not 100% sure, in the documentation I found it's supposed to return the last time step available with that appended, but even like this I only get 1 return value which seems correct to me

Reply:
hm I think RF may not read correctly this line of code. In the html log I see:
`${expression} = DCGM_FI_PROF_GR_ENGINE_ACTIVE`

Reply:
oh I thought the "#" was supposed to be part of the query, sorry :)

Reply:
Removed it since it's not needed currently